### PR TITLE
[ui] Asset view support for column lineage metadata

### PR DESCRIPTION
**Pull request by bengotow**
## Summary & Motivation

This PR adds a new "Column" selector to the Asset > Lineage page for assets that emit `lineage` materialization event metadata. Choosing a column switches you to a column-oriented version of the graph, which supports the same upstream/downstream scoping + navigation as the normal lineage view, but shows column-level arrows and info.

This PR makes a few changes to our asset layout system in order to support this new DAG styling:

- The asset layout engine takes a wider range of it's hardcoded constants as configuration (previously just `direction`), and you can override Dagre layout settings. This allows us to run asset layout (with the worker, indexdb caching, etc), but with smaller `column` nodes. The options were already part of the cache keys, so this all works pretty smoothly!

- I found an open PR on Dagre that fixes the issue with asset groups overlapping occasionally because the layout engine could not account for the presence of the "title bar" we render on them!  I added the adjustments to our existing `dagre.patch` file.  This is important in this PR because the nodes are the columns and the groups are the assets, so they overlap badly without this fix.
 
## How I Tested These Changes

image
image
image

### CODE REVIEW

File: js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetNode.tsx
Comment (bengotow) [inline js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetNode.tsx:39]: At some point this changed and the assets now shift up when there are no reason tags, which makes the arrows misaligned

Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetNode.tsx:39]: oof thats on me. It changed last week!

Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetNode.tsx:39]: Should we change it back?

Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetNode.tsx:39]: Ahh it's ok! I think adding this min-height does the trick 🙏

File: js_modules/dagster-ui/packages/ui-core/src/assets/SavedZoomLevel.tsx
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/assets/SavedZoomLevel.tsx:30]: You didn't actually use the assetGraphId here? I imagine the intent was to persist the zoom levels of different graphs independently?

File: js_modules/dagster-ui/packages/ui-core/src/assets/lineage/useColumnLineageDataForAssets.tsx
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/assets/lineage/useColumnLineageDataForAssets.tsx:86]: feels bad to have to parse this again to avoid the hook warning

File: js_modules/dagster-ui/packages/ui-core/src/assets/lineage/useColumnLineageDataForAssets.tsx
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/assets/lineage/useColumnLineageDataForAssets.tsx:88]:
```
if (fetching.current) {
  return;
}
```

File: js_modules/dagster-ui/packages/ui-core/src/assets/SavedZoomLevel.tsx
Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/assets/SavedZoomLevel.tsx:30]: Ahh interesting... let me dig into this a bit, I think we may have been trying to make sure the zoom resets to the saved zoom when you select assets

**Comment by bengotow** [comment]
Changed the base of this PR to master, it's not technically blocked on the backend, the metadata just won't appear until that lands.

**Review by salazarm — APPROVED** [review]

File: js_modules/dagster-ui/packages/ui-core/src/assets/lineage/useColumnLineageDataForAssets.tsx
Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/assets/lineage/useColumnLineageDataForAssets.tsx]: Ahh good call, I memoized a few things upstream and that allows us to just pass in the original `missing` array. Thanks!

File: js_modules/dagster-ui/packages/ui-core/src/assets/SavedZoomLevel.tsx
Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/assets/SavedZoomLevel.tsx]: It looks like that was the case - i renamed this var so it's more obvious what it's doing here!

File: js_modules/dagster-ui/packages/ui-core/src/assets/lineage/useColumnLineageDataForAssets.tsx
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/assets/lineage/useColumnLineageDataForAssets.tsx:82]: Should this use LiveDataProvider for the batching/chunking polling benefits?

**Pull request merged**
